### PR TITLE
MWPW-161117 consumer selection

### PR DESCRIPTION
**Pull request by npeltier**
- cards are displayed from top folder corresponding to shown tabs,
- only CCD is default, more can be displayed if added to  local storage,
- remove of useless breadcrumb,
- didn't fix table view padding as not sure we'll be using it yet
image

Test URLs:
- Before: https://main--mas--adobecom.hlx.live/studio.html
- After: https://MWPW-161117--mas--adobecom.hlx.live/studio.html

### CODE REVIEW

**Review by 3ch023 — CHANGES_REQUESTED** [review]
both #path (opening a tab) and #query=nala (search) are using the hash, this seems to be the reason this PR is breaking search.
Maybe we switch to query parameters instead of hash?

**Comment by npeltier** [comment]
> Maybe we switch to query parameters instead of hash?
thanks i'll have a look. Either way is good, catalog is using hash with different dimensions, but i agree query parameters looks better to me cc @yesil

**Comment by npeltier** [comment]
> > Maybe we switch to query parameters instead of hash?
> 
> thanks i'll have a look. Either way is good, catalog is using hash with different dimensions, but i agree query parameters looks better to me cc @yesil

discussed it internally with @3ch023 & @yesil and we agreed to keep things this way as we have other stuff to do than reimplementing the wheel atm :)

**Comment by 3ch023** [comment]
https://mwpw-161117--mas--adobecom.hlx.live/studio.html#path=%2Fcontent%2Fdam%2Fmas%2Fnala shows no results

**Comment by yesil** [comment]
there is too much white space above the tabs. 
image

I think we need to move filter panel buttons inside content-navigation and have only a single toolbar.

image

File: studio/src/aem/content-navigation.js
Comment (npeltier) [reply studio/src/aem/content-navigation.js:80]: "problem" with using lit-html is that it implies adding tabs to shadow DOM. I'm fine trying that, but we agreed offline to add them to light DOM

**Comment by npeltier** [comment]
> I think we need to move filter panel buttons inside content-navigation and have only a single toolbar.

this would need longer discussion with PM/UX, i propose not to do anything on that topic in that PR

**Comment by npeltier** [comment]
@3ch023 @yesil  please have a second look. I know styling is still not perfect, but as discussed we are still not sure about the overall styling itself.

**Comment by npeltier** [comment]
image
this is easily achievable though moving `<mas-filter-toolbar></mas-filter-toolbar>` to content-navigation if you prefer (i don't have opinions)

**Comment by 3ch023** [comment]
seems like we agreed offline to do the dropdown instead of tabs

File: studio/src/aem/content-navigation.js
Comment (yesil) [inline studio/src/aem/content-navigation.js:100]: name suggestions: `get roots`

File: studio/test/aem/aem-fragments.test.js
Comment (yesil) [inline studio/test/aem/aem-fragments.test.js:1]: 👍

**Review by yesil — APPROVED** [review]

File: studio/src/aem/content-navigation.js
Comment (yesil) [reply studio/src/aem/content-navigation.js:100]: I found consumer too opinionated, because, in my idea, everything under aem folder should remain generic and reusable.

File: studio/src/aem/aem-fragments.js
Comment (3ch023) [inline studio/src/aem/aem-fragments.js:182]: `this.#searchResult = [];` already happened in line 173, no?

File: studio/src/aem/aem-fragments.js
Comment (3ch023) [inline studio/src/aem/aem-fragments.js:185]: this needs an await, maybe you could add it, otherwise it should come with Ilyases changes

File: studio/src/aem/aem-fragments.js
Comment (3ch023) [inline studio/src/aem/aem-fragments.js:197]: this is needed for setting correct hash when you search by fragment id

File: studio/src/aem/content-navigation.js
Comment (npeltier) [reply studio/src/aem/content-navigation.js:100]: i can keep top folders

Comment (npeltier) [reply studio/src/aem/content-navigation.js:100]: but i guess i like the binding of top folders to consumers/surfaces: i agree that nala/docs/drafts surfaces are strange, but we could still consider them this way.
Essentially they are top folders with limited write access, and that is how we designed our consumer/surface storage

File: studio/src/aem/aem-fragments.js
Comment (npeltier) [reply studio/src/aem/aem-fragments.js:182]: no, this is here in case some fragment is found but you are in wrong folder

**Review by 3ch023 — APPROVED** [review]

File: studio/src/aem/mas-filter-toolbar.js
Comment (yesil) [inline studio/src/aem/mas-filter-toolbar.js:23]: do we need it as an attribute?  As you user changes the search text, we would end up having an inconsistency with that is in the attribute and what is in the search. If we use `reflect:true`, that it will cause unnecessary DOM updates.

File: studio/src/aem/mas-filter-toolbar.js
Comment (yesil) [inline studio/src/aem/mas-filter-toolbar.js:35]: cannot this get the searchText value from deeplink?
also, I propose to drop the `mas-` prefix in the file names. It is already the MAS repository.

File: studio/src/aem/content-navigation.js
Comment (yesil) [inline studio/src/aem/content-navigation.js:122]: ` this.source?.removeEventListener(EVENT_LOAD, this.handlerSourceLoad());` will not remove the listener.

I recommend to bind the handler in the constructor and then, add and remove it as follows:
```js
this.source?.addEventListener(EVENT_LOAD, this.handlerSourceLoad);
this.source?.removeEventListener(EVENT_LOAD, this.handlerSourceLoad);
```

Comment (yesil) [reply studio/src/aem/content-navigation.js:122]: so that the same instance is added and removed.

File: studio/src/aem/content-navigation.js
Comment (yesil) [inline studio/src/aem/content-navigation.js:5]: I think `getTopFolder` should be in a different file.
since `aem-fragments.js` doesn't belong to content-navigation, it should not be directly imported IMO.

File: studio/src/aem/mas-filter-toolbar.js
Comment (npeltier) [reply studio/src/aem/mas-filter-toolbar.js:23]: this is used for init

File: studio/src/aem/mas-filter-toolbar.js
Comment (npeltier) [reply studio/src/aem/mas-filter-toolbar.js:35]: i could, but this also triggers a event handler creation i'm not a fan of here

**Comment by npeltier** [comment]
@yesil thanks for your comments, let's treat them as part as planned refactor

**Comment by yesil** [comment]
@npeltier I think we should at least correct the even listeners before merging.

**Review by yesil — APPROVED** [review]

**Comment by afmicka** [comment]
@npeltier the search does not work for me if i use `aem.env=qa` ... as soon as i press enter, the `aem.env` parameter gets removed from the url (that does not happen if you click Search button) 
1. go to https://mwpw-161117--mas--adobecom.hlx.live/studio.html?aem.env=qa
2. enter text to search
3. click enter

**Comment by afmicka** [comment]
@npeltier it is also not performing search with partial word. If i search with the word 'photo' and not 'photoshop' i get zero results. Shouldn't i get all the cards that have photoshop in it? 
Screenshot 2024-11-06 at 16 57 22

**Comment by afmicka** [comment]
@npeltier there is a regression in the preview on the cards. Any change in the RTE fields is not previewed on the card. Open any card, make a change in eny of the fields (title, badge, description), click outside the field and notice the change is not previewed.

**Review by Axelcureno — APPROVED** [review]

**Comment by Axelcureno** [comment]
Looks neat 🚀

**Comment by afmicka** [comment]
@npeltier as agreed, created jira with the made observation for future improvments https://jira.corp.adobe.com/browse/MWPW-161894

the rest looks good to me. Approving the PR

**Pull request merged**
